components/patient-data/ui/src/main/resources/PhenoTips/AllergiesDataClass.xml

```diff
@@ -72,7 +72,7 @@
   #if($xcontext.action == 'edit' || $xcontext.action == 'inline')
     {{html wiki=false clean=false}}&lt;input class="xwiki-free-multiselect-value" type="text" size="30" name="${prefix}${name}" id="${prefix}${name}_${counter}" value="$!escapetool.xml($v)"/&gt;{{/html}}##
   #else
-    $escapetool.xml($v)##
+    $!{services.rendering.escape($v, $doc.syntax)}##
```

Comment:
Files on disk don't support wiki markup, they just pass through Velocity and are copied to the output as-is. Content in wiki pages are passed through the rendering engine, and Velocity code in them must explicitly be placed inside `{{velocity}}` macros.

What do you mean by "custom escape tool"? `$escapetool` is the general escaping tool to be used, and it has useful methods for escaping different kinds of content (xml, javascript, json, url parameters, email headers). Ideally, it should also have a method for escaping wiki markup, but that's not possible for several reasons:

- The escape tool is defined in xwiki-commons, while the rendering engine is in xwiki-rendering, which is an upstream project depending on xwiki-commons, and the base project can't depend on the upstream, otherwise it would cause a circular dependency
- the rendering engine is a component, while Velocity tools are plain objects

In theory it is possible to define an "enhanced escapetool" and use it in PhenoTips, but I think it's too much effort for something that already works.